codec_record.go

```diff
@@ -76,9 +76,17 @@ func decoderOfStruct(d *decoderContext, schema Schema, typ reflect2.Type) ValDec
 				}
 			}
 		}
-
-		// Skip field if it doesnt exist
+		// Skip field if it doesn't exist
 		if sf == nil {
+			// If the field value doesn't exist in the binary, ignore it instead of
+			// appending a 'SkipDecoder'.
+			//
+			// Note: 'SkipDecoder' performs a read and moves the cursor, which,
+			// in this case, will lead to a dirty read.
+			if field.action == FieldSetDefault {
```

Comment:
I would have expected in this case that we do ignore the field, but also set the default if it exist? Is that incorrect?

Reply:
We do set the default if the corresponding struct field exists. This is done a bit further down in the code.

I think a struct with a missing corresponding field left us with two options:
1. skip the field value and move the reader's cursor forward
2. ignore if the field value doesn't exist in the binary in the first place (the case of a field with FieldSetDefault flag)

Let me know if I didn’t fully answer the question.

Reply:
But you continue on the next line, so we never set the default decoder?

Reply:
Right. I get it now.

Reply:
No, I knew what it was doing, I somehow had the idea it was in the reader. As you point out, it is not in the writer or the reader, but rather an extra field in the struct.

Reply:
No, I messed that up, it is not in the writer and not in the struct, but in the reader.